From: =?utf-8?q?Jonas_=C3=85dahl?= <jadahl@gmail.com>
Date: Fri, 15 Jan 2021 10:26:01 +0100
Subject: backend/native: Destroy logind helper after seat

Otherwise we don't actually close input devices using it, since that
would happen after meta_launcher_free() was called.

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1670>
---
 src/backends/native/meta-backend-native.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/backends/native/meta-backend-native.c b/src/backends/native/meta-backend-native.c
index c794891..d2af33b 100644
--- a/src/backends/native/meta-backend-native.c
+++ b/src/backends/native/meta-backend-native.c
@@ -101,9 +101,10 @@ meta_backend_native_dispose (GObject *object)
 
   g_clear_object (&native->udev);
   g_clear_object (&native->kms);
-  g_clear_pointer (&native->launcher, meta_launcher_free);
 
   G_OBJECT_CLASS (meta_backend_native_parent_class)->dispose (object);
+
+  g_clear_pointer (&native->launcher, meta_launcher_free);
 }
 
 static ClutterBackend *
